oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticIndexWriter.java

```diff
@@ -253,16 +260,36 @@ public void afterBulk(long executionId, BulkRequest bulkRequest, BulkResponse bu
                     LOG.error("Error decoding bulk response", e);
                 }
             }
+            NodeBuilder status = definitionBuilder.child(IndexDefinition.STATUS_NODE);
             if (bulkResponse.hasFailures()) { // check if some operations failed to execute
+
+                // Read the current failed paths (if any) on the :status node into failedDocList
+                if (status.getProperty(IndexDefinition.FAILED_DOC_PATHS) != null) {
+                    for (String str : status.getProperty(IndexDefinition.FAILED_DOC_PATHS).getValue(Type.STRINGS)) {
+                        failedDocSet.add(str);
+                    }
+                }
+
                 for (BulkItemResponse bulkItemResponse : bulkResponse) {
                     if (bulkItemResponse.isFailed()) {
                         BulkItemResponse.Failure failure = bulkItemResponse.getFailure();
-                        LOG.error("Bulk item with id {} failed", failure.getId(), failure.getCause());
+                        failedDocSet.add(bulkItemResponse.getId());
+                        // Log entry to be used to parse logs to get the failed doc id/path if needed
+                        LOG.error("ElasticIndex Update Doc Failure: Error while adding/updating doc with id : [{}]", bulkItemResponse.getId());
+                        LOG.error("Failure Details: BulkItem ID: {}, Failure Cause: {}", failure.getId(), failure.getCause());
```

Comment:
This will not treat `failure.getCause()` as an exception. The method signature with a Throwable parameter is - 
```void error(String var1, Throwable var2);```
You should use - 
``` LOG.error("Failure Details: BulkItem ID: " + failure.getId() + ", Failure Cause: {}", failure.getCause());```